modules/jaeger-thrift-http-exporter/src/main/scala/io/janstenpickle/trace4cats/jaeger/JaegerHttpSpanExporter.scala

```diff
@@ -3,41 +3,59 @@ package io.janstenpickle.trace4cats.jaeger
 import cats.Foldable
 import cats.effect.kernel.Async
 import cats.syntax.either._
+import cats.syntax.flatMap._
 import cats.syntax.foldable._
 import cats.syntax.functor._
+import fs2.{Chunk, Stream}
 import io.jaegertracing.thriftjava.{Batch => JaegerBatch, Process, Span}
 import io.janstenpickle.trace4cats.`export`.HttpSpanExporter
 import io.janstenpickle.trace4cats.kernel.SpanExporter
 import io.janstenpickle.trace4cats.model.{Batch, TraceProcess}
 import org.apache.thrift.TSerializer
+import org.http4s._
 import org.http4s.client.Client
 import org.http4s.headers.`Content-Type`
-import org.http4s.{Header, MediaType, Uri}
+
+import scala.util.control.NonFatal
 
 object JaegerHttpSpanExporter {
 
-  private val headers: List[Header.ToRaw] = List(`Content-Type`(MediaType.application.`vnd.apache.thrift.binary`))
+  private val thriftBinary: List[Header.ToRaw] = List(`Content-Type`(MediaType.application.`vnd.apache.thrift.binary`))
+
+  private implicit def jaegerBatchEncoder[F[_]: Async](implicit
```

Comment:
Is `Async` needed here, or can we get away with a slightly less powerful typeclass?